include/util/io.hpp

```diff
@@ -66,17 +56,14 @@ bool serializeVector(std::ostream &stream, const std::vector<simple_type> &data)
 template <typename simple_type>
 bool deserializeVector(const std::string &filename, std::vector<simple_type> &data)
 {
-    std::ifstream stream(filename, std::ios::binary);
 
-    if (!readAndCheckFingerprint(stream))
-        return false;
+    storage::io::File file(filename, true);
```

Comment:
Meh, can you please remove the defaulted boolean and add an enum class at least

`enum class CheckFingerprint { Yes, No };`

no idea what `true` means for a File in this context without taking a look at the File header.